exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/Table.kt

```diff
@@ -882,6 +882,17 @@ open class Table(name: String = "") : ColumnSet(), DdlAware {
         _indices.add(Index(columns.toList(), isUnique, customIndexName))
     }
 
+    /**
+     * Creates an index with a custom index type.
+     *
+     * @param customIndexName Name of the index.
+     * @param columns Columns that compose the index.
+     * @param isUnique Whether the index is unique or not.
+     */
+    fun index(customIndexName: String? = null, indexType: String, vararg columns: Column<*>) {
```

Comment:
Maybe it's better to extend existing `index` function with `indexType: String? = null`. It should not break existing code. 
But maybe it's not possible to create unique index with custom type?

Reply:
The limitations of custom index types are not very well standardized.
 - Postgres only supports unique indexes for `BTREE`
 - MySQL supports unique indexes using `HASH` and `BTREE`, but multi-column indexes must be `BTREE`
 - SQLite does not have any custom index types

Reply:
I went ahead and addedthis to the existing `index` function.

I do not believe there is a clean, vendor-agnostic API that describes exactly when custom index types are available. Perhaps it is best to provide the option and fail at runtime if the database implementation does not support the specified index.